packages/record-data/addon/-private/relationships/state/has-many.ts

```diff
@@ -162,25 +170,9 @@ export default class ManyRelationship extends Relationship {
     for (let i = 0, l = recordDatas.length; i < l; i++) {
       this.addCanonicalRecordData(recordDatas[i], i);
     }
-  }
-
-  setInitialRecordDatas(recordDatas: RelationshipRecordData[] | undefined) {
-    if (Array.isArray(recordDatas) === false || !recordDatas || recordDatas.length === 0) {
-      return;
-    }
-
-    for (let i = 0; i < recordDatas.length; i++) {
-      let recordData = recordDatas[i];
-      if (this.canonicalMembers.has(recordData)) {
-        continue;
-      }
 
-      this.canonicalMembers.add(recordData);
-      this.members.add(recordData);
-      this.setupInverseRelationship(recordData);
-    }
-
-    this.canonicalState = this.canonicalMembers.toArray();
+    // sync with local state
+    this.flushCanonicalLater();
```

Comment:
This was the main change - at the end of `computeChanges`, flush to sync with local state